### DISCUSSION TITLE
Adding an "except" method into InteractsWithProperties.php

### DISCUSSION THREAD

**Discussion by nihad-ziyatkhanli** [livewire/livewire#3704] [2. Show and tell]
Hello, there!
Since component has `only` method, I think it needs `except` method to be compatible with laravel's array helpers and collection methods.
During development I needed to access all/most public properties, and searching in docs there were no methods other than `only`, which made me type all the properties manually. I opened src files and saw there is a method called `getPublicPropertiesDefinedBySubClass`, which would return all public properties I listed, but there was no except method. So I  implemented it myself and would open a pull request, if you approve below changes.

Preferred way:
```
public function except($properties)
{
   return array_diff_key($this->getPublicPropertiesDefinedBySubClass(), array_flip($properties));
}
```
Alternative:
```
use Illuminate\Support\Arr;

public function except($properties)
{
   return Arr::except($this->getPublicPropertiesDefinedBySubClass(), $properties);
}
```
Also for the sake of compatibility and making it short for users, `all` method can be registered as alias:
```
public function all()
{
   return $this->getPublicPropertiesDefinedBySubClass();
}
```
So what do you think?

**Comment by nihad-ziyatkhanli** [comment]
Should I make a pull request? Has been 2 weeks.

**Reply by joshhanley** [reply]
@nihad-ziyatkhanli I personally don't have a need, but if you've already written the code, then PR's are welcome. Just be sure to add an example of code usage to the PR. 🙂
